### ISSUE TITLE
Agenda integration broken when not using daily files

### ISSUE THREAD

**Issue by lhernanz** [bastibe/org-journal#296]
**Describe the bug**
There is a bug with date calculations when you are not using the daily type. The problem is on the method `org-journal--update-org-agenda-files`. For the daily case, the method is subtracting one for picking up the current day file. But for the case of monthly or yearly, it should get back to the previous month/year so that the search function picks up the current file. 

More specifically, the offending line seems to be
```
(setcar (cdr beg) (1- (cadr beg))) ;; Include today; required for `org-journal--search-build-file-list'
 
```

The problem is that `org-journal--file-name->calendar-date` maps monthly journals to the first day of the month (and the yearly ones to the first day of the year), so they are never picked up by the above function.

**To Reproduce**
Change to any type of journal that is not daily. Enable agenda integration and check that no file is added to the org-agenda-files variable

**Expected behavior**
The current agenda file should be added to the variable so that it is included in the agenda items.

**Screenshots**
N/A

**Desktop (please complete the following information):**
 - Emacs 27.

**Your Emacs Configuration**
N/A

**Comment by cslux** [comment]
Hmm, works for!

**Comment by cslux** [comment]
Please try it again with the latest MELPA unstable version. The yearly journal file is even included for past scheduled items for me, see image. 
![Screenshot from 2020-09-12 15-50-26](https://user-images.githubusercontent.com/1712313/92996979-0a575980-f510-11ea-8030-d2998f0be1bf.png)

**Comment by lhernanz** [comment]
Hi,

it is still not working for me with monthly files and by naming them only using yyyymm.org.

I might still be completely wrong about this, but I still thing that the offending line is the one that I have highlighted before in the ```org-journal--update-org-agenda-files``` function. 

The problem is that I would get the following execution for today (October 1st):
```
(org-journal--search-build-file-list
              (org-journal--calendar-date->time '(10 00 2020))
              (org-journal--calendar-date->time '(12 31 2030)))
``` 
That returns an empty list. In order for the code to pick up the right file, I would need to subtract from the month field instead of from the day field (in reality any date in September would work for me).
```
(org-journal--search-build-file-list
              (org-journal--calendar-date->time '(09 01 2020))
              (org-journal--calendar-date->time '(12 31 2030)))
``` 
This produces the current journal file: 202010.org.

I am not sure why is it working for you with the yearly example. I am not sure if the fact that you are including the day as part of the name of the files makes the difference or not.

Regards

**Comment by anupam312nwd** [comment]
Hi @lhernanz,

Is your issue resolved? I am looking for a solution of this too, since I am using `org-journal-file-type` as `monthly` too, and integration of org-journal with org-agenda is not working for me. It was working I think when I was using `weekly` file-type.

Thanks

**Comment by lhernanz** [comment]
It is still not working for me. This is the reason that I stated in my last
message that the bug is still there IMHO. I am not sure how much visibility
a closed issue will get. I am not sure if we should open a new one.

**Comment by anupam312nwd** [comment]
Thanks @lhernanz for your reply. Thanks @bastibe for reopening it.

**Comment by onnyyonn** [comment]
I can confirm that the filename matters in this issue. When I have `org-journal-file-format "%Y-%m-%d.org"` for my monthly journal, agenda integration works fine. But with `org-journal-file-format "%Y-%m.org"`, tasks don't show up in agenda.

**Comment by anupam312nwd** [comment]
> I can confirm that the filename matters in this issue. When I have `org-journal-file-format "%Y-%m-%d.org"` for my monthly journal, agenda integration works fine. But with `org-journal-file-format "%Y-%m.org"`, tasks don't show up in agenda.

Changing `org-journal-file-format` to any of `%Y-%m-%d.org` or `%Y-%m-%d` is working.

**Comment by cslux** [comment]
Should be fixed with fa7be5f. You figured it already out! The bug was in `org-journal--file-name->calendar-date`. In the case where the month or day was missing it was set to 0 instead of 1.

**Comment by anupam312nwd** [comment]
@lhernanz is org journal integration working for you? org journal integration is not working for me with both `%Y-%m-%d.org` and `%Y-%m.org` format.
